codalab/server/authenticated_plugin.py

```diff
@@ -0,0 +1,54 @@
+from bottle import (
+  abort,
+  httplib,
+  redirect,
+  request,
+  url,
+)
+
+
+class AuthenticatedPlugin(object):
+    """
+    Fails the request if the user is not authenticated (i.e. request.user hasn't
+    been set).
+
+    This method redirects the user to login unless the request is an AJAX request
+    (i.e. has X-Requested-With header set to XMLHttpRequest). The worker system
+    sets this header as well and so should any other code that doesn't want
```

Comment:
It's not just the worker system...it's also the CLI in the future.  In the documentation, let's be more explicit (bulleted list) about all the use cases.